### PR TITLE
Adding two eggs to include automatic installation of GeyserMC and Floodgate for Spigot and BungeeCord

### PR DESCRIPTION
**Pull request by Greaper88**
Modification of the official install scripts for Spigot and BungeeCord to include installing/updating GeyserMC and floodgate. Exported from Panel version 0.7.18.

### CODE REVIEW

File: egg-bungeecord--geyser--floodgate.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
+    "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PTERODACTYL PANEL - PTERODACTYL.IO",
+    "meta": {
+        "version": "PTDL_v1"
+    },
+    "exported_at": "2020-11-23T02:52:35-06:00",
+    "name": "Bungeecord+Geyser+Floodgate",
+    "author": "support@pterodactyl.io",
```
Comment (rtm516) [inline egg-bungeecord--geyser--floodgate.json:8]:
```suggestion
    "name": "Bungeecord + Geyser + Floodgate",
    "author": "support@geysermc.org",
```

File: egg-spigot--geyser--floodgate.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
+{
+    "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PTERODACTYL PANEL - PTERODACTYL.IO",
+    "meta": {
+        "version": "PTDL_v1"
+    },
+    "exported_at": "2020-11-23T02:51:12-06:00",
+    "name": "Spigot+Geyser+Floodgate",
+    "author": "support@pterodactyl.io",
```
Comment (rtm516) [inline egg-spigot--geyser--floodgate.json:8]:
```suggestion
    "name": "Spigot + Geyser + Floodgate",
    "author": "support@geysermc.org",
```

File: egg-bungeecord--geyser--floodgate.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    "description": "For a long time, Minecraft server owners have had a dream that encompasses a free, easy, and reliable way to connect multiple Minecraft servers together. BungeeCord is the answer to said dream. Whether you are a small server wishing to string multiple game-modes together, or the owner of the ShotBow Network, BungeeCord is the ideal solution for you. With the help of BungeeCord, you will be able to unlock your community's full potential.",
```
Comment (rtm516) [inline egg-bungeecord--geyser--floodgate.json:9]: Could this be altered to mention geyser and floodgate get installed?

File: egg-spigot--geyser--floodgate.json [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    "description": "Spigot is the most widely-used modded Minecraft server software in the world. It powers many of the top Minecraft server networks around to ensure they can cope with their huge player base and ensure the satisfaction of their players. Spigot works by reducing and eliminating many causes of lag, as well as adding in handy features and settings that help make your job of server administration easier.",
```
Comment (rtm516) [inline egg-spigot--geyser--floodgate.json:9]: Could this be altered to mention geyser and floodgate get installed?

File: egg-spigot--geyser--floodgate.json [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            "script": "#!\/bin\/ash\r\n# Spigot Installation Script\r\n#\r\n# Server Files: \/mnt\/server\r\n\r\napk add --no-cache --update curl git openjdk8 openssl bash nss\r\n\r\n## Only download if a path is provided, otherwise continue.\r\nif [ ! -z \"${DL_PATH}\" ]; then\r\n    cd \/mnt\/server\r\n\r\n    MODIFIED_DOWNLOAD=`eval echo $(echo ${DL_PATH} | sed -e 's\/{{\/${\/g' -e 's\/}}\/}\/g')`\r\n    curl -sSL -o ${SERVER_JARFILE} ${MODIFIED_DOWNLOAD}\r\nelse\r\n    cd \/srv\/\r\n    \r\n    wget https:\/\/hub.spigotmc.org\/jenkins\/job\/BuildTools\/lastSuccessfulBuild\/artifact\/target\/BuildTools.jar\r\n    \r\n    mv BuildTools.jar \/srv\/\r\n\r\n    java -jar BuildTools.jar --rev ${DL_VERSION}\r\n\r\n    mv spigot-*.jar \/mnt\/server\/${SERVER_JARFILE}\r\n    \r\n    curl -o \/mnt\/server\/plugins\/Geyser-Spigot.jar https:\/\/ci.nukkitx.com\/job\/GeyserMC\/job\/Geyser\/job\/master\/lastSuccessfulBuild\/artifact\/bootstrap\/spigot\/target\/Geyser-Spigot.jar\r\n    \r\n    curl -o \/mnt\/server\/plugins\/floodgate-bukkit.jar https:\/\/ci.nukkitx.com\/job\/GeyserMC\/job\/Floodgate\/job\/master\/lastSuccessfulBuild\/artifact\/bukkit\/target\/floodgate-bukkit.jar\r\n\r\nfi",
```
Comment (rtm516) [inline egg-spigot--geyser--floodgate.json:20]: Could the default config be downloaded and combined with the use of `sed` so it enables floodgate out of the box?

File: egg-bungeecord--geyser--floodgate.json [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        "startup": "{\r\n    \"done\": \"Listening on \",\r\n    \"userInteraction\": [\r\n        \"Listening on \/0.0.0.0:25577\"\r\n    ]\r\n}",
+        "logs": "{\r\n    \"custom\": false,\r\n    \"location\": \"proxy.log.0\"\r\n}",
+        "stop": "end"
+    },
+    "scripts": {
+        "installation": {
+            "script": "#!\/bin\/ash\r\n# Bungeecord Installation Script\r\n#\r\n# Server Files: \/mnt\/server\r\napk update\r\napk add curl\r\n\r\ncd \/mnt\/server\r\n\r\nif [ -z \"${BUNGEE_VERSION}\" ] || [ \"${BUNGEE_VERSION}\" == \"latest\" ]; then\r\n    BUNGEE_VERSION=\"lastStableBuild\"\r\nfi\r\n\r\ncurl -o ${SERVER_JARFILE} https:\/\/ci.md-5.net\/job\/BungeeCord\/${BUNGEE_VERSION}\/artifact\/bootstrap\/target\/BungeeCord.jar\r\ncurl -o plugins\/Geyser-BungeeCord.jar https:\/\/ci.nukkitx.com\/job\/GeyserMC\/job\/Geyser\/job\/master\/lastSuccessfulBuild\/artifact\/bootstrap\/bungeecord\/target\/Geyser-BungeeCord.jar\r\ncurl -o plugins\/floodgate-bungee.jar https:\/\/ci.nukkitx.com\/job\/GeyserMC\/job\/Floodgate\/job\/master\/lastSuccessfulBuild\/artifact\/bungee\/target\/floodgate-bungee.jar",
```
Comment (rtm516) [inline egg-bungeecord--geyser--floodgate.json:20]: Could the default config be downloaded and combined with the use of `sed` so it enables floodgate out of the box?

**Review by rtm516 — CHANGES_REQUESTED** [review]

File: egg-bungeecord--geyser--floodgate.json (hunk 3 above)
Comment (Greaper88) [reply egg-bungeecord--geyser--floodgate.json:9]: lemme take a look, and i'll re-export them

File: egg-spigot--geyser--floodgate.json (hunk 5 above)
Comment (Greaper88) [reply egg-spigot--geyser--floodgate.json:20]: how would i go about doing that?

File: egg-bungeecord--geyser--floodgate.json [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    },
+    "scripts": {
+        "installation": {
+            "script": "#!\/bin\/ash\r\n# Bungeecord Installation Script\r\n#\r\n# Server Files: \/mnt\/server\r\napk update\r\napk add curl\r\n\r\ncd \/mnt\/server\r\n\r\nif [ -z \"${BUNGEE_VERSION}\" ] || [ \"${BUNGEE_VERSION}\" == \"latest\" ]; then\r\n    BUNGEE_VERSION=\"lastStableBuild\"\r\nfi\r\n\r\ncurl -o ${SERVER_JARFILE} https:\/\/ci.md-5.net\/job\/BungeeCord\/${BUNGEE_VERSION}\/artifact\/bootstrap\/target\/BungeeCord.jar\r\ncurl -o plugins\/Geyser-BungeeCord.jar https:\/\/ci.nukkitx.com\/job\/GeyserMC\/job\/Geyser\/job\/master\/lastSuccessfulBuild\/artifact\/bootstrap\/bungeecord\/target\/Geyser-BungeeCord.jar\r\ncurl -o plugins\/floodgate-bungee.jar https:\/\/ci.nukkitx.com\/job\/GeyserMC\/job\/Floodgate\/job\/master\/lastSuccessfulBuild\/artifact\/bungee\/target\/floodgate-bungee.jar\r\nFILE=plugins\/Geyser-BungeeCord\/config.yml\r\nif [[ ! -f \"plugins\/Geyser-BungeeCord\/config.yml\" ]]; then\r\n  curl -o plugins\/Geyser-BungeeCord\/config.yml https:\/\/raw.githubusercontent.com\/GeyserMC\/Geyser\/master\/connector\/src\/main\/resources\/config.yml\r\n  sed -i 's\/: online\/: floodgate\/g' config.yml\r\nfi",
```
Comment (rtm516) [inline egg-bungeecord--geyser--floodgate.json:20]:
```suggestion
            "script": "#!\/bin\/ash\r\n# Bungeecord Installation Script\r\n#\r\n# Server Files: \/mnt\/server\r\napk update\r\napk add curl\r\n\r\ncd \/mnt\/server\r\n\r\nif [ -z \"${BUNGEE_VERSION}\" ] || [ \"${BUNGEE_VERSION}\" == \"latest\" ]; then\r\n    BUNGEE_VERSION=\"lastStableBuild\"\r\nfi\r\n\r\ncurl -o ${SERVER_JARFILE} https:\/\/ci.md-5.net\/job\/BungeeCord\/${BUNGEE_VERSION}\/artifact\/bootstrap\/target\/BungeeCord.jar\r\ncurl -o plugins\/Geyser-BungeeCord.jar https:\/\/ci.nukkitx.com\/job\/GeyserMC\/job\/Geyser\/job\/master\/lastSuccessfulBuild\/artifact\/bootstrap\/bungeecord\/target\/Geyser-BungeeCord.jar\r\ncurl -o plugins\/floodgate-bungee.jar https:\/\/ci.nukkitx.com\/job\/GeyserMC\/job\/Floodgate\/job\/master\/lastSuccessfulBuild\/artifact\/bungee\/target\/floodgate-bungee.jar\r\n\r\nif [[ ! -f \"plugins\/Geyser-BungeeCord\/config.yml\" ]]; then\r\n  curl -o plugins\/Geyser-BungeeCord\/config.yml https:\/\/raw.githubusercontent.com\/GeyserMC\/Geyser\/master\/connector\/src\/main\/resources\/config.yml\r\n  sed -i 's\/: online\/: floodgate\/g' config.yml\r\nfi",
```

File: egg-bungeecord--geyser--floodgate.json [hunk 8]
```diff
@@ -0,0 +1,45 @@
+{
+    "_comment": "DO NOT EDIT: FILE GENERATED AUTOMATICALLY BY PTERODACTYL PANEL - PTERODACTYL.IO",
+    "meta": {
+        "version": "PTDL_v1"
+    },
+    "exported_at": "2020-11-23T04:04:40-06:00",
+    "name": "Bungeecord+Geyser+Floodgate",
```
Comment (rtm516) [inline egg-bungeecord--geyser--floodgate.json:7]:
```suggestion
    "name": "Bungeecord + Geyser + Floodgate",
```

**Review by rtm516 — CHANGES_REQUESTED** [review]

File: egg-bungeecord--geyser--floodgate.json (hunk 7 above)
Comment (Greaper88) [reply egg-bungeecord--geyser--floodgate.json:20]: what is the change here? I

**Comment by Greaper88** [comment]
Closing this PR to open a new one using Paper MC instead of Spigot.

**Pull request closed without merging**